node/service/src/relay_chain_selection.rs

```diff
@@ -304,6 +309,20 @@ where
 		self.block_header(best_leaf)
 	}
 
+	async fn finality_target(
+		&self,
+		_target_hash: Hash,
+		_maybe_max_number: Option<BlockNumber>,
+	) -> Result<Option<Hash>, ConsensusError> {
+		unreachable!("Must call finality_target_with_fallback instead. qed");
```

Comment:
This exists in a trait implementation which is called by the GRANDPA code. Wouldn't that unconditionally hit the panic in this case?

Reply:
no, we only pass a wrapping type which contains both `selection` and `fallback`. This wrapper will be removed once the overseer proc macro gains the ability to use closures for initialization and avoid the `connected` /`disconnected` states.

Hence the unreachable.

Reply:
I'd prefer to just remove the unused trait implementation and move the other methods to inherent impl

Reply:
Yeah, you are correct. The initial idea of just removing `struct RelayChainSelWithFallback` is obsolete now that we need fallback unconditionally. 👍